ExampleMod/Content/Projectiles/ExampleWhipProjectile.cs

```diff
@@ -0,0 +1,168 @@
+			Projectile.spriteDirection = Projectile.velocity.X >= 0f ? 1 : -1;
+
+			// This statement handles charging. The maximum charge time is doubled to accommodate for the projectile's extra update. If you don't want this functionality, replace the statement with just Timer++;
+			if (owner.channel && ChargeTime < 120) {
+				ChargeTime++;
+			}
+			else {
+				Timer++;
+			}
+
+			Projectile.GetWhipSettings(Projectile, out float timeToFlyOut, out int _, out float _);
+
+			if (Timer >= timeToFlyOut || owner.itemAnimation <= 0) {
+				Projectile.Kill();
+				return;
+			}
+
+			owner.heldProj = Projectile.whoAmI;
+
+			// These two lines ensure that the timing of the owner's use animation is correct.
+			owner.itemAnimation = owner.itemAnimationMax - (int)(Timer / Projectile.MaxUpdates);
+			owner.itemTime = owner.itemAnimation;
+
+			if (Timer == timeToFlyOut / 2) {
+				// Plays a whipcrack sound at the tip of the whip.
+				List<Vector2> points = Projectile.WhipPointsForCollision;
+				Projectile.FillWhipControlPoints(Projectile, points);
+				SoundEngine.PlaySound(SoundID.Item153, points[points.Count - 1]);
+			}
+		}
+
+		public override void GetWhipSettings(Player player, ref float timeToFlyOut, ref int segments, ref float rangeMultiplier) {
+			segments = 20 + (int)ChargeTime / 12;
+			rangeMultiplier = 1 + (int)ChargeTime / 120f;
+		}
+
+		public override void OnHitNPC(NPC target, int damage, float knockback, bool crit) {
+			target.AddBuff(ModContent.BuffType<ExampleWhipDebuff>(), 240);
+			Main.player[Projectile.owner].MinionAttackTargetNPC = target.whoAmI;
+		}
+
+		// This method draws a line between all points of the whip, in case there's empty space between the sprites.
+		private void DrawLine(List<Vector2> list) {
+			Texture2D texture = TextureAssets.FishingLine.Value;
+			Rectangle frame = texture.Frame();
+			Vector2 origin = new Vector2(frame.Width / 2, 2);
+
+			Vector2 pos = list[0];
+			for (int i = 0; i < list.Count - 1; i++) {
+				Vector2 element = list[i];
+				Vector2 diff = list[i + 1] - element;
+
+				float rotation = diff.ToRotation() - MathHelper.PiOver2;
+				Color color = Lighting.GetColor(element.ToTileCoordinates(), Color.White);
+				Vector2 scale = new Vector2(1f, (diff.Length() + 2) / frame.Height);
+
+				Main.EntitySpriteDraw(texture, pos - Main.screenPosition, frame, color, rotation, origin, scale, SpriteEffects.None, 0);
+
+				pos += diff;
+			}
+		}
+
+		public override bool PreDraw(ref Color lightColor) {
+			List<Vector2> list = new List<Vector2>();
+			Projectile.FillWhipControlPoints(Projectile, list);
+
+			DrawLine(list);
+
+			// If you don't want to use custom code, you can instead call one of vanilla's DrawWhip methods. However, keep in mind that you must adhere to how they draw if you do.
+			// Main.DrawWhip_WhipBland(Projectile, list);
```

Comment:
Please make it more obvious that all the following code is for Custom Draw - A simple comment preceding the rest of the code that everything to follow is Custom Draw would go a long ways for people who skip this line.

Also, very minor, but recommended comment styling is:
// Comment
//code
So can visually see what is comments without reading.